### PR TITLE
Organize all resgroup tests to the same directory

### PR DESCRIPTION
**Pull request by pengzhout**
resource group tests are messing up isolation2 directory, this commits
move them all to the same directory to be tidy and organized.it also
    handle the sub directory in input/output directory for resgroup

### CODE REVIEW

File: src/test/regress/pg_regress.c [hunk 1]
```diff
@@ -624,6 +625,106 @@ generate_uao_sourcefiles(char *src_dir, char *dest_dir, char *suffix, replacemen
...
+		char		srcfile[MAXPGPATH];
+		char		destfile[MAXPGPATH];
+		char		prefix[MAXPGPATH];
+		FILE	   *infile,
+				   *outfile;
+		char		line[1024];
+		bool		has_tokens = false;
+
+		/* reject filenames not finishing in ".source" */
+		if (strlen(*name) < 8)
```
Comment (kainwen) [inline src/test/regress/pg_regress.c]: Maybe `char *ext = strrchr(str, '.'); ` then check `ext == NULL or strcmp(ext, ".source") != 0` is a little easier to understand.

Comment (pivotal-ning-yu) [inline src/test/regress/pg_regress.c]: Current implementation in this PR looks fine to me as it's the same with the `generate_uao_sourcefiles()` function in the same file. Suppose we decide to optimize the logic I would suggest we also do the same to `generate_uao_sourcefiles()`, or even better, merge them into one function.

**Comment by pivotal-ning-yu** [comment]
LGTM

**Comment by pengzhout** [comment]
@kainwen @pivotal-ning-yu simplified the logic, updated.

**Comment by pengzhout** [comment]
Merged, thanks for review.

**Pull request merged**